azurerm/data_source_app_service_virtual_network_connection.go

```diff
@@ -0,0 +1,150 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmAppServiceVirtualNetworkConnection() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAppServiceVirtualNetworkConnectionRead,
+
+		Schema: map[string]*schema.Schema{
+			"app_service_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAppServiceName,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"virtual_network_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Could we validate this with at least `NoEmptyStrings` (and we might have an existing vnet name validator from that resource?)